src/@next/Tag/Tag.tsx

```diff
@@ -0,0 +1,25 @@
+import React from 'react';
+import { Typography } from '../Typography';
+import { Neutral } from '../utilities/colors';
+import { TagStyle } from './TagStyle';
+
+export interface TagProps extends React.HTMLAttributes<HTMLDivElement> {
+  children?: React.ReactNode;
+}
+
+export const getTagChildren = (children: React.ReactNode) => {
+  if (typeof children !== 'string') {
+    console.warn('Tag children should only accept string');
```

Comment:
hmm, i beg to differ on this

it would be wise at component level to determine what kind of icon to render by passing in the necessary attributes; something like what [polaris tag](https://polaris.shopify.com/components/tag) is doing

Reply:
if we allow user to clump whatever they want inside the Tag component, the tag component is gonna look very weird. i think we also set a standard of how much customization the component should have

Reply:
If you mean to reference Polaris, they allow more flexibility in terms of what consuming apps can render as children though (the `with custom content` example they have) which is what I'm suggesting. From the standard we currently have, we strictly type the props and check whether there's a specific composition within the component but we allow flexibility on what consuming apps can compose if the design allows them to. But of course, for the close button and links for example, we'll be controlling these.

Reply:
got it @ninariccimarie 
fixed, please check again